doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownParser.java

```diff
@@ -156,6 +178,86 @@ public void parse(Reader source, Sink sink, String reference) throws ParseExcept
         }
     }
 
+    private boolean processMetadataForHtml(StringBuilder html, StringBuilder source) {
+        final Map<String, List<String>> metaData;
+        final int endOffset; // end of metadata within source
+        // support two types of metadata:
+        if (source.toString().startsWith("---")) {
```

Comment:
Empty lines before are not allowed?

Reply:
No (although no formal spec, this is not supported by most parsers, see also https://github.com/jonschlinkert/gray-matter/issues/71#issuecomment-382723455).